src/native/fs/fs_napi.cpp

```diff
@@ -931,13 +1011,25 @@ struct Readfile : public FSWorker
             remain -= len;
             p += len;
         }
+
+        CHECK_CTIME_CHANGE(fd, _stat_res, _path);
```

Comment:
Can it happen that a race happens after the comparison of ctime and before function exit? Is this interleaving possible?

Reply:
Yes but the "exit" from the function is not a point of interest, what we care here is that ctime has not changed while we were reading the content and the xattrs of the file, which means we read an "atomic" state of the file. After we completed loading everything to memory, it can change, but our memory holds a consistent state of the file. Hope this is clear.